utils/to_string.cc

```diff
@@ -10,39 +10,54 @@
 
 namespace std {
 
-std::ostream& operator<<(std::ostream& os, const std::strong_ordering& order) {
+std::ostream& boost_test_print_type(std::ostream& os, const std::strong_ordering& order) {
```

Comment:
This is for testing, isn't it? In that case, it's better to localize it in test/ directory

Reply:
yes, they are. i am moving them into `test/lib/test_utils.{hh,cc}`. does this make more sense now?

Reply:
Yes, it's much better now, thank you.

Now I have out-of-curiosity question (not a merge blocked) -- why do we need those helpers at all? Boost uses operator<< to print types and doesn't "see" the newly introduced formatters, right?

Reply:
yeah, Boost.test does not use {fmt} or even std::format, it uses either operator<< and falls back to  `boost_test_print_type()`, if the former is not available. actually, i didn't have to go through the fuss of renaming `operator<<` to `boost_test_print_type()` and moving them to their new home. but i really want to make it explicit that these helpers are for testing, and not supposed to be a part of our formatting system.